### ISSUE TITLE
Documentation: add DDBBs documentation section for ReadThedocs.io

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#920]
In GitLab by @ jberlin on Dec 16, 2022, 12:20

related to https://earth.bsc.es/gitlab/digital-twins/de_340/project_management/-/issues/216

The idea is to provide a brief description (summary) of the Autosubmit databases based on the work done for the documentation of the Autosubmit/Autosubmit GUI/Autosubmit API DDBBs, It should be short and focused on explaining the kinds of databases (distributed and central), location (the experiment directory, the as_metadata and the directory for the central DDBBs, explained if the location is parametrized in the configuration), general purpose for each one of the three, and who writes those

Original documentation:
https://docs.google.com/document/d/10KoXpY3fy1-zCCOIMIYmmbWhnTWj-3BZ6szgU_QfnmA/edit#

@mcastril @dbeltrankyl @manuel-g-castro

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 16, 2022, 12:21

mentioned in issue digital-twins/de_340/project_management#216

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 16, 2022, 18:09

Here is the first draft for a summary we can add:

https://docs.google.com/document/d/1lJ_8iJel8LZMZvPtaHuJsND8TXuOiGzHqC1JgWt2hHs/edit

I made it in a googledoc so we can easily comment and discuss

**Comment by kinow** [comment]
Thanks for writing these docs. We had an issue in Marenostrum this week, and I immediately knew about the different databases, how they were used, and that helped to me diagnose the issue (even though I couldn't fix it). I believe these docs will also help newcomer and even experienced users.

>I made it in a googledoc so we can easily comment and discuss

I think the link was shared with access as Viewer for users with the link (which I think might be good given these issues are public), so commenting here:

>central databases:
>ecearth.db

I imagine that the readers of these new ReadTheDocs docs will be devs & users from ECMWF, AWI, CSC, etc. Probably members of the DestinE project, or other external users.IMHO it would be better to use the default value here, and explain the that is the default value, but it can be customized. Otherwise `ecearth.db` is confusing for readers if they tried Autosubmit 4 and have only seen an `autosubmit.db` in their own directories.

>Databases stored in /esarchive/autosubmit/as_metadata

Ditto the above. Except for BSC devs, other readers would not understand what is `/esarchive/autosubmit`.

I liked how you separated the auxiliary databases, and also how you added a note to each database description saying who writes, and who reads to/from it :clap: (I will use a similar approach to document databases in the future).

I thought about having that text to be displayed like this, more or less, for the readers:

---

Autosubmit stores information about its experiments and workflows in SQLite databases and as serialized Python objects (pickle files). There are two types of databases in Autosubmit: central databases, and auxiliary databases. There are also [log files](some page describing logging in AS) with important information about experiments (statuses, timestamps, messages, [error codes](link to error codes page)).

## Central Databases

- **autosubmit.db**: Autosubmit's main database. The default name is `autosubmit.db`, but name and location can customized in `.autosubmitrc`. Written and read by Autosubmit.
- **as_times.db**: Used by the [Autosubmit API](link to autosubmit api repo/docs?). This database is deprecated database since 3.x, kept for backward compatibility. Written and read by the `populate_times.py` [worker](link to docs about workers?).

## Auxiliary Databases

(*The types of auxiliary databases as you described :thumbsup: ...*)

## Python objects

(*Describe the pickle files here, and summarize why we have that; It could be a surprise to some users that we use that, or at least it was to me as I was expecting everything to be serialized into SQLite entities :) *)

---

That's just a suggestion. I think the most important is to make the docs for the DestinE users thinking that they have no access to Marenostrum, and they might be trying that in their own laptops or LUMI, Leonardo, etc.

Thanks!!

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 19, 2022, 10:47

Hi Bruno, sorry I thought I gave proper access rights, I just fixed it:

https://docs.google.com/document/d/1lJ_8iJel8LZMZvPtaHuJsND8TXuOiGzHqC1JgWt2hHs/edit?usp=sharing

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 19, 2022, 10:54

Thanks for the feedback @kinow , I fully agree with make the documentation HPC agnostic

**Comment by kinow** [comment]
In GitLab by @mcastril on Dec 19, 2022, 11:52

> Hi Bruno, sorry I thought I gave proper access rights, I just fixed it:
> 
> https://docs.google.com/document/d/1lJ_8iJel8LZMZvPtaHuJsND8TXuOiGzHqC1JgWt2hHs/edit?usp=sharing

I think it's better not to give edition rights and only commenters instead, as issues are public, as Bruno said.

**Comment by kinow** [comment]
In GitLab by @mcastril on Dec 19, 2022, 11:52

> Thanks for the feedback @kinow , I fully agree with make the documentation HPC agnostic

Note that this had been already commented in https://earth.bsc.es/gitlab/es/autosubmit/-/issues/842#note_191216

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 19, 2022, 11:53

I just changed it (the link)

**Comment by kinow** [comment]
I think the text in the document can be used to craft one, maybe two paragraphs for the document, with a link for the page in RTD once it's deployed there. I will start working on this over the next days, just ping me whenever you have the text in a merge request or available in RTD, please, @ jberlin :thumbsup: 

From there we can discuss in a MR or in a new issue refinements to improve it, while in parallel we keep working on the rejected milestone.

Thanks!

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 19, 2022, 16:57

Hi @kinow , I followed some of the suggestions you mentioned above, now I divided the databases into CORE ones and Auxiliary ones added some more information and made the suggestions about making this HPC agnostic, please feel free to have a look.

**Comment by kinow** [comment]
Excellent work @ jberlin !! I cannot even remember where we had the ecearth.db :-) I left a few suggestions, feel free to accept/reject. Nothing major in my suggestions & comments.

IMO, we could include the text as in your document and that would already be a good start for users (and for our DestinE document :tada: ). We can iterate more later after we give time for others to read it in the RTD pages.

Thanks heaps for updating it so quickly!

**Comment by kinow** [comment]
In GitLab by @mcastril on Dec 19, 2022, 19:53

Good progress, I added some suggestions in the doc.

**Comment by kinow** [comment]
I left a comment about the distributed sqlite suggestion, but +1 to all others! Thanks @mcastril !

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 20, 2022, 18:00

Hi @mcastril, @kinow I applied most of the corrections remarked, the only one that remains, is the one of the beginning, I made a suggestion on a possible writing in the comment, thanks to both of you

**Comment by kinow** [comment]
It's looking good @ jberlin . We can have more iterations on this document later, but I think we have enough to publish it, and add a link in the DestinE docs. I left a couple suggestions (a reply to your comment too), but feel free to accept/reject as you see fit. My +1 :thumbsup: thanks!

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 21, 2022, 10:14

Hi @kinow , sounds good !

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 21, 2022, 16:08

Hello, 

@ jberlin, @kinow  check https://earth.bsc.es/gitlab/digital-twins/de_340/project_management/-/issues/216#note_192397

If you have doubts about how to upload the db doc into readthedocs. @manuel-g-castro can also help. 

Just to mention that the changes must be added to the master branch. 

But if you have to remake the v4.0.0b ( to also have the doc update on there ) I put the instructions there just in case but I guess you already know how to do it. 

1. Go to https://earth.bsc.es/gitlab/es/autosubmit/-/tags 
2. Delete v4.0.0b
3. Press on "New Tag" 
4. In New tag field: v4.0.0b
5. In Create from field: master
6. Scroll down and create tag. 

About the module version, since it is only documentation there is no need to upload a new module. 

If you need to, you have to talk with avilamir ( or maybe pabretonniere, or mcastril ) to request access as **eadmin** in bscearth000 (I'm avoiding to tag them to avoid spam ) 

then `ssh eadmin@ bscearth000` and execute `/shared/earth/ComputationalES/eadmin/easybuild/prepare_autosubmit4.sh`

After you did the recreation of the v4.0.0b tag.

Also, for autosubmit 3.14 fixes if any, is the same procedure but 

then `ssh eadmin@ bscearth000` and execute `/shared/earth/ComputationalES/eadmin/easybuild/prepare_autosubmit.sh`

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 21, 2022, 17:42

I also made several changes to original [source document](https://docs.google.com/document/d/10KoXpY3fy1-zCCOIMIYmmbWhnTWj-3BZ6szgU_QfnmA/edit#) with the idea of making the documentation agnostic to the platform

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 22, 2022, 17:31

mentioned in commit 8bf3fcf6546aedb4364f8d8987b5166bc6f11a56

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 22, 2022, 17:40

I created !297 for putting this initial version to readthedocs.io

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 27, 2022, 12:51

mentioned in commit e12a637f0ca9ae0acd5898e1d7d2b44bb8475abb

**Comment by kinow** [comment]
MR merged, now do we close this issue @ jberlin ? (my first MR merged :medal: )

**Comment by kinow** [comment]
In GitLab by @ jberlin on Dec 27, 2022, 13:11

Hi @kinow , I think we can close it since it is now merged, thanks for reviewing and merging the changes

**Comment by kinow** [comment]
Perfect. Thank **you** for the great new docs. I'm updating the milestone that will link to this new page :slight_smile: :thumbsup:
